src/main/java/com/google/sps/data/User.java

```diff
@@ -17,11 +30,26 @@ public User(MapLocation geolocation) {
   * @param zipcode
   * @return User with zipCode
   **/
-  public User(int zipCode) {
-
+  public User(String zipCode) {
+    this.geolocation = zipCodeToMapLocation(zipCode);
   }
 
   public MapLocation getGeolocation() {
     return geolocation;
   }
+
+  private MapLocation zipCodeToMapLocation(String zipCode) {
+    final GeoApiContext context = new GeoApiContext.Builder()
+            .apiKey(KEY)
+            .build();
+    final GeocodingResult[] results;
+    try {
+        results = GeocodingApi.geocode(context, zipCode).await();
+        MapLocation geolocation = new MapLocation(results[0].geometry.location.lat, results[0].geometry.location.lng);
```

Comment:
Should this only care about the first element in the results?

Reply:
The API documentation says that generally only one result is returned unless the input is particularly ambiguous. It seems as though every zipCode that I have tried only returns one result.

Reply:
I couldn't the docs for this method? Does it provide a getFirst() call or similar?

Reply:
Not that I know of. This is the class that the method is from: https://github.com/googlemaps/google-maps-services-java/blob/master/src/main/java/com/google/maps/GeocodingApi.java

I was just referencing the location by the first element in the array. Does that answer your question?

Reply:
Yes. But looking at this method, it doesn't seem to return multiple objects. Just one.

Reply:
I think that the .await() triggers the getResult() function (along with the others) at the end of that file which does return an array of results. I tried to make the type just a GeocodingResult in my code and I got an incompatible types error.

Reply:
Gotcha. I see now. Thanks for clarifying this.